numpy/lib/financial.py

```diff
@@ -209,11 +218,12 @@ def pmt(rate, nper, pv, fv=0, when='end'):
     when = _convert_when(when)
     (rate, nper, pv, fv, when) = map(np.array, [rate, nper, pv, fv, when])
     temp = (1 + rate)**nper
-    mask = (rate == 0.0)
-    masked_rate = np.where(mask, 1.0, rate)
```

Comment:
Using `1` instead of `1.0` would be enough here

Reply:
I guess first pass with these changes going with integers should be fine.  The reason I was a bit more dynamic was an attempt to support other types, than only decimal. But first pass I should probably only do decimal.

Reply:
I figure that any type that knows how to add itself to integers is enough. I can't come up with any financially useful type that wouldn't allow that.